airflow/datasets/manager.py

```diff
@@ -96,12 +96,11 @@ def _slow_path_queue_dagruns(self, dataset: DatasetModel, session: Session) -> N
     def _postgres_queue_dagruns(self, dataset: DatasetModel, session: Session) -> None:
         from sqlalchemy.dialects.postgresql import insert
 
-        stmt = insert(DatasetDagRunQueue).values(dataset_id=dataset.id).on_conflict_do_nothing()
-        session.execute(
-            stmt,
-            [{'target_dag_id': target_dag.dag_id} for target_dag in dataset.consuming_dags],
-        )
-        session.flush()
+        params = [{'target_dag_id': target_dag.dag_id} for target_dag in dataset.consuming_dags]
+        if params:
+            stmt = insert(DatasetDagRunQueue).values(dataset_id=dataset.id).on_conflict_do_nothing()
+            session.execute(stmt, params)
+            session.flush()
```

Comment:
Change looks good but should we add a test case for it?

Reply:
It also only cover one path, there are two.

Reply:
Updated with test and a simpler place to make the change.

Reply:
the other path wasn't susceptible to the issue